hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/federation/FederationStateStoreService.java

```diff
@@ -283,154 +290,229 @@ public void checkVersion() throws Exception {
   @Override
   public GetSubClusterPolicyConfigurationResponse getPolicyConfiguration(
       GetSubClusterPolicyConfigurationRequest request) throws YarnException {
-    return stateStoreClient.getPolicyConfiguration(request);
+    FederationClientMethod clientMethod = new FederationClientMethod(
```

Comment:
Wouldn't it make more sense to make it:
```
FederationClientMethod<GetSubClusterPolicyConfigurationResponse>
```
?

Reply:
This is a good idea, I will modify the code.